### PR TITLE
fix(api-server): stop migration from resurrecting deleted connections

### PR DESCRIPTION
**Pull request by PetrBulanek**
## Problem

The #2200 migration defers deleting the legacy `platform-cred-*` Secrets to #2198, so they linger. The migration recreates a connection for any legacy Secret without one — so a connection the user **deletes gets rebuilt on the next api-server boot**. Users can't remove migrated connections.

## Fix

Only (re)create a connection **while an agent still grants the legacy Secret**. Once flipped, the Secret is ungranted, so a deleted connection stays deleted. Still-granted Secrets migrate and re-flip as before; an ungranted Secret with no connection (deleted, or an unused orphan) is logged and left for the #2198 sweep.

## Validated (local, full pre-cutover → migrate → delete → restart)

Fresh migration still drains (`migrated 2, failed 0`); after deleting a connection and restarting, it logged `skip … no granting agent` and **did not reappear**.

Interim fix for the #2200 → #2198 window. #2198 removes the migration and must also sweep the retained Secrets. Refs #1273, #2198

### CODE REVIEW

File: packages/api-server/src/modules/connections/migration/secrets-to-connections.ts [hunk 1]
```diff
@@ -207,6 +212,24 @@ async function ensureConnection(
...
 
+  // Only (re)create a connection while an agent still grants the legacy secret.
+  // Once the first pass flips an agent, the legacy secret is ungranted — so a
+  // connection the user later deletes must NOT be rebuilt. Because deletion is
+  // deferred to #2198, the legacy Secret lingers, and a bare "no connection
+  // yet" check would resurrect the deleted connection on every boot (or the
+  // 60s retry). An ungranted secret with no connection is either that
+  // user-deleted case or an unused orphan; log and leave it for the #2198 sweep
+  // rather than migrate it into a connection nothing uses.
+  if (!existing && granting.length === 0) {
```
Comment (dam-code-guardian) [inline packages/api-server/src/modules/connections/migration/secrets-to-connections.ts:223]: 🟢 **Suggestion:** This resurrection guard is exactly the kind of boot-migration edge case worth locking down with a regression test (pre-cutover → migrate → delete the connection → restart → assert it is not recreated). The PR body says this was validated locally; there's no dedicated test file for `secrets-to-connections.ts` in the repo, so the guard currently has no automated coverage. Not blocking.

**Review by dam-code-guardian — APPROVED** [review]
🛡️ **DAM** — ✅ Code Review @ `229863f`

## PR #2337: fix(api-server): stop migration from resurrecting deleted connections
**Author:** PetrBulanek | **Branch:** fix/2200-migration-no-resurrect → main | **Changes:** +23 −0 (1 file)

### Summary
Adds a guard in `ensureConnection()` so the legacy-secrets→connections boot migration only (re)creates a connection while an agent still grants the legacy secret. Because secret deletion is deferred to #2198, the legacy Secret lingers; without this guard a connection the user deletes would be rebuilt on the next api-server boot (or the 60s retry). An ungranted secret with no connection (user-deleted, or an unused orphan) is now logged and left for the #2198 sweep.

### Findings
- ✅ **Looks good:** The guard is correctly ordered — it sits *after* `recoverValue()` (so an unrecoverable value still takes the earlier `value unrecoverable` skip) and *before* the dry-run/create branches, so no connection is ever created for an ungranted-and-absent credential. (`secrets-to-connections.ts:223`)
- ✅ **Looks good:** Returning `null` routes through the existing phase-1 `if (!res) { skipped++; continue; }` path, so the skip is counted and reported consistently with the other permanent skips; no agent-grant mutation happens. (`secrets-to-connections.ts:142`)
- ✅ **Looks good:** `findGrantingAgents()` unions across all of `cred.secretIds` (primary + twins), so a PAT/Bob pair where either half is still granted is correctly treated as "still granted" and migrates as before.
- 🟢 **Suggestion:** This resurrection guard is exactly the kind of boot-migration edge case that benefits from a regression test (pre-cutover → migrate → delete connection → restart → assert not recreated), which the PR body says was validated locally. There is no dedicated test file for `secrets-to-connections.ts` in the repo, so the behavior is currently only covered by manual validation. (`secrets-to-connections.ts:223`)

### Documentation Check (doc-drift)
✅ No documentation drift detected.

### Verdict
APPROVE — Small, well-reasoned defensive fix that closes a real user-facing bug (deleted connections resurrecting) with correct ordering and no behavioral regression for still-granted secrets; the only note is the absence of an automated regression test.

---
_Review by [DAM](https://github.com/dam-agents/dam) · automated code guardian_

**Review by jezekra1 — APPROVED** [review]
